titus-supplementary-component/task-relocation/src/main/java/com/netflix/titus/supplementary/relocation/evacuation/EvacuationExecutor.java

```diff
@@ -0,0 +1,215 @@
+package com.netflix.titus.supplementary.relocation.evacuation;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.netflix.titus.api.agent.model.AgentInstance;
+import com.netflix.titus.api.agent.model.AgentInstanceGroup;
+import com.netflix.titus.api.agent.model.InstanceOverrideState;
+import com.netflix.titus.api.eviction.model.EvictionQuota;
+import com.netflix.titus.api.jobmanager.TaskAttributes;
+import com.netflix.titus.api.jobmanager.model.job.Job;
+import com.netflix.titus.api.jobmanager.model.job.JobFunctions;
+import com.netflix.titus.api.jobmanager.model.job.Task;
+import com.netflix.titus.api.jobmanager.model.job.TaskState;
+import com.netflix.titus.api.model.reference.Reference;
+import com.netflix.titus.runtime.connector.agent.AgentDataReplicator;
+import com.netflix.titus.runtime.connector.eviction.EvictionDataReplicator;
+import com.netflix.titus.runtime.connector.eviction.EvictionDataSnapshot;
+import com.netflix.titus.runtime.connector.eviction.EvictionServiceClient;
+import com.netflix.titus.runtime.connector.jobmanager.JobDataReplicator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class EvacuationExecutor {
```

Comment:
Some tests for this class would be :100:

Reply:
I expect this code to be discarded as soon as I start working on a more generic solution evaluating multiple criteria. Writing tests with all these interfaces would be difficult without up front investment into the test toolkit. I plan to do this in the phase 2.